src/backend/license-list/client.lambda-shared.ts

```diff
@@ -0,0 +1,59 @@
+import { s3 } from '../shared/aws.lambda-shared';
+import { requireEnv } from '../shared/env.lambda-shared';
+import { EnvironmentVariables } from './constants';
+
+/**
+ * A client for working with the license list.
+ */
+export class LicenseListClient {
+  public static async newClient() {
+    const client = new LicenseListClient();
+    await client.init();
+    return client;
+  }
+
+  readonly #bucketName: string;
+  readonly #objectKey: string;
+
+  #map?: Map<string, string>;
+
+  private constructor() {
+    this.#bucketName = requireEnv(EnvironmentVariables.BUCKET_NAME);
+    this.#objectKey = requireEnv(EnvironmentVariables.OBJECT_KEY);
+  }
+
+  /**
+   * Looks up a license ID in this list.
+   *
+   * @param licenseId the license ID to look up (possibly with unorthodox case).
+   *
+   * @returns the normalized SPDX license identifier for this license, if it is
+   *          in the list, or `undefined` otherwise.
+   */
+  public lookup(licenseId: string): string | undefined {
+    /* istanbul ignore if (should never happen) */
+    if (this.#map == null) {
```

Comment:
same suggestion as above
```suggestion
    if (this.#map === undefined) {
```

Reply:
You talked about type coercion before... I do _not_ like coercing values to a boolean context when what I am trying to assert is whether the value is defined or not. `false`-y in JavaScript can be extremely surprising, so I prefer not to depend on it anywhere (i.e: "is-defined" checks look the same everywhere in the code, as opposed to being different depending on the type of the value being asserted).